database/migrations/2020_05_01_121000_create_order_items_table.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Tipoff\Checkout\Models\Order;
+use Tipoff\Checkout\Models\OrderItem;
+
+class CreateOrdersTable extends Migration
+{
+    public function up()
+    {
+        Schema::create('order_items', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(Order::class);
+            $table->foreignIdFor( OrderItem::class, 'parent_id')->nullable();
+
+            // TODO - TBD - change to Type=>class mapping instead of Morph??  using morph forces the sellable to be a Model!
+            $table->morphs('sellable');
+
+            // Opaque item identifier provided by Sellable
+            $table->string('item_id');
+            // User friendly description for cart line item
+            $table->string('description');
+            $table->unsignedInteger('quantity');
+
+            // Pair is handled as cast to `DiscountableValue`
+            // TODO - TBD - is this price each?
+            $table->unsignedInteger('amount')->default(0);
+            $table->unsignedInteger('amount_discounts')->default(0);
```

Comment:
With `quantity` as part of cart item, I wasn't sure if the intent was for these to be price for each?  or the total for the item?

Reply:
@drewroberts - would like to get this resolved.  Within a cart or order item, does amount the price each?  or the total price for the given quantity?  (wasn't an issue for escape rooms when quantity was always 1)

Reply:
I'm still thinking about this. Leaning towards using `amount` for the price of each item. Would that require an additional field or at least an accessor for the line item amount?

Maybe use `amount_per_item` as the field name to avoid confusion and also have an `amount` field or accessor that is the total for that cart item entry with quantity taken into effect.

Reply:
From my experience, if there's only one stored value its the price per item.  Just didn't want to force that on you without your input.  If that ends up being your preference, I'd lean towards keeping both sets of fields in the record (amount_per_item/amount_per_item_discounts and the existing amount/amount_discounts).  The latter pair will always be a computation based on the former * quantity.  

The only real change would be the API to create new items would expect a per item price.  (And I'd update them to be clear).  Also, line item dollar off discounts would still operate off the total (5 @ $10 w/ $10 discount => $40, not free).

Reply:
Sounds good. Let me think on this a little more.